components/autocomplete/index.js

```diff
@@ -330,11 +338,25 @@ export class Autocomplete extends Component {
 		// create a regular expression to filter the options
 		const search = open ? new RegExp( '(?:\\b|\\s|^)' + escapeRegExp( query ), 'i' ) : /./;
 		// filter the options we already have
-		const filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
+		let filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
 		// check if we should still suppress the popover
 		const suppress = ( open && wasSuppress === open.idx ) ? wasSuppress : undefined;
 		// update the state
 		if ( wasOpen || open ) {
+			if ( open ) {
+				if ( this.props.completers[ open.idx ].setSearch ) {
+					this.props.completers[ open.idx ].setSearch( query, this.props.completers[ open.idx ].getOptions ).then( ( options ) => {
+						const keyedOptions = map( options, ( option, i ) => ( { ...option, key: open.idx + '-' + i } ) );
```

Comment:
The original intent of adding the key here was that each unique option would have a unique key that was not effected by the filtering - this effects the ARIA the most because it used the key to determine if the selection had changed and needed to be re-announced. Unfortunately the dynamic server-side search breaks this assumption and I am not sure how you will fix it.

Reply:
Hmm, not sure i understand.  the issue is the tool will not re announce the selection, or that it will over announce when the selection hasn't changed?

Reply:
could we use the item slug here which should be unique?

Reply:
I reworked the keying to use slugs if available, seems to work but needs testing

Reply:
Ok, make sure it works in the general case (ie for the slash completer and the proposed hash-tag completer).